### ISSUE TITLE
Remove METRICS dictionary

### ISSUE THREAD

**Issue by craffel** [craffel/mir_eval#54]
tl;dr - the METRICS dictionary doesn't really serve the purpose I want it to, and the evaluators do, so it shouldn't exist.

In some cases, metrics within the same task might have a different number of arguments.  This used to be the case in `segment` before it got split into `boundary` and `structure`.  It's currently true in `melody`; `voicing_measures` just takes in voicing arrays while the rest take in voicing and frequency arrays (even though `raw_pitch_accuracy` and `raw_chroma_accuracy` don't actually use their `est_voicing` arg).  If all metrics in each task submodule took the same arguments, you could run

``` Python
for name, metric in mir_eval.task_submodule.METRICS.items():
    print name, metric(ref, est)
```

or whatever the call signature is, which would be convenient.  But, we don't expect this to be the case in general.  This was kind of the main reason for the METRICS dictionary to exist, and in practice the evaluators submodule should essentially subsume this functionality.
